src/java.base/share/classes/jdk/internal/foreign/layout/SequenceLayoutImpl.java

```diff
@@ -179,7 +179,7 @@ public SequenceLayout flatten() {
 
     @Override
     public String toString() {
-        boolean max = (Long.MAX_VALUE / elementLayout.byteSize()) == elemCount;
+        boolean max = (Long.MAX_VALUE / Math.max(1, elementLayout.byteSize())) == elemCount;
```

Comment:
Looks good - now if you have a sequence layout whose count is Long.MAX_VALUE, but whose element layout has size 0, you also get the `*` in the toString, which I guess makes sense (as you can't have a bigger sequence layout than that).